### PR TITLE
[Windows] Add android NDK r23

### PR DESCRIPTION
**Pull request by nikita-bykov**
# Description
In scope of this PR we have added Android NDK r23 to Windows images and leave the current LTS version (r21) as default.
Also, we've reworked the `android.ndk` toolset section:

The `android.ndk.versions` field is a manually sorted list of all versions that will be installed on the image
The `android.ndk.lts` field has been renamed to `android.ndk.default`. It will allow us, if necessary, to set a non-LTS NDK version as default.
The `android.ndk.latest` field has been removed and its value will be taken from the end of `android.ndk.versions` list

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated: [Windows2016](https://github.visualstudio.com/virtual-environments/_build/results?buildId=116591&view=results), [Windows2019](https://github.visualstudio.com/virtual-environments/_build/results?buildId=116592&view=results), [Windows2022](https://github.visualstudio.com/virtual-environments/_build/results?buildId=116590&view=results)

### CODE REVIEW

File: images/win/scripts/Installers/Install-AndroidSDK.ps1 [hunk 1]
```diff
@@ -109,21 +111,28 @@ $ndkLatestPackageName = Get-AndroidPackagesByName -AndroidPackages $androidPacka
                 | Sort-Object -Unique `
                 | Select-Object -Last 1
 
-$androidNDKs = @($ndkLTSPackageName, $ndkLatestPackageName)
+$androidNDKs = @()
```
Comment (al-cheb) [inline images/win/scripts/Installers/Install-AndroidSDK.ps1:114]: Let's use Foreach-Object
```
$androidNDKs = $ndkMajorVersions | Foreach-Object {
    Get-AndroidPackagesByName -AndroidPackages $androidPackages -PrefixPackageName "ndk;$_" | Sort-Object -Unique | Select-Object -Last 1
}
```

File: images/win/scripts/Tests/Android.Tests.ps1 [hunk 2]
```diff
@@ -5,8 +5,13 @@ Describe "Android SDK" {
     $androidPackages = Get-AndroidPackages -AndroidSDKManagerPath (Get-AndroidSDKManagerPath)
     $androidInstalledPackages = Get-AndroidInstalledPackages
 
-    $ndkLTSMajorVersion = $androidToolset.ndk.lts
-    $ndkLatestMajorVersion = $androidToolset.ndk.latest
+    $ndkDefaultMajorVersion = $androidToolset.ndk.default
+    $ndkLatestMajorVersion = $androidToolset.ndk.versions | Select-Object -Last 1
+    $ndkPackagesTestCases = @()
+    $ndkVersions = $androidToolset.ndk.versions
+    $ndkVersions | ForEach-Object {
```
Comment (al-cheb) [inline images/win/scripts/Tests/Android.Tests.ps1:12]:
```
$ndkPackagesTestCases = $ndkVersions | ForEach-Object {
    @{ ndkPackage = $_; installedPackages = $androidInstalledPackages }
}
```

File: images/win/scripts/SoftwareReport/SoftwareReport.Android.psm1 [hunk 3]
```diff
@@ -158,6 +158,28 @@ function Get-AndroidGoogleAPIsVersions {
...
+function Get-AndroidNdkVersions {
+    param (
+        [Parameter(Mandatory)]
+        [object] $PackageInfo
+    )
+
+    $ndkLinkTarget = (Get-Item $env:ANDROID_NDK_HOME).Target
+    $ndkDefaultFullVersion = Split-Path -Path $ndkLinkTarget -Leaf
+    
+    $versions = $packageInfo | Where-Object { $_ -Match "ndk;" } | ForEach-Object {
```
Comment (al-cheb) [inline images/win/scripts/SoftwareReport/SoftwareReport.Android.psm1:170]: Difficult to understand, let's short:
```
$versions = $packageInfo | Where-Object { $_ -Match "ndk;" } | ForEach-Object {
    $version = (Split-TableRowByColumns $_)[1]
    if ($_ -eq $ndkDefaultFullVersion) {
        $version += " (default)"
    }
    $packageInfoParts
}
$version -join "<br>"
```

File: images/win/scripts/Installers/Install-AndroidSDK.ps1 (hunk 1 above)
Comment (nikita-bykov) [reply images/win/scripts/Installers/Install-AndroidSDK.ps1]: Fixed it

File: images/win/scripts/Tests/Android.Tests.ps1 (hunk 2 above)
Comment (nikita-bykov) [reply images/win/scripts/Tests/Android.Tests.ps1]: Fixed it

File: images/win/scripts/SoftwareReport/SoftwareReport.Android.psm1 (hunk 3 above)
Comment (nikita-bykov) [reply images/win/scripts/SoftwareReport/SoftwareReport.Android.psm1:170]: Fixed it

File: images/win/scripts/Tests/Android.Tests.ps1 [hunk 4]
```diff
@@ -91,8 +95,12 @@ Describe "Android SDK" {
...
         }
 
-        It "LTS NDK is installed" -TestCases @(@{ ndkLTSVersion = $ndkLTSMajorVersion; installedPackages = $androidInstalledPackages }) {
-            "$installedPackages" | Should -Match "ndk;$ndkLTSVersion"
+        It "NDK <ndkPackage> is installed" -TestCases $ndkPackagesTestCases {
+            "$installedPackages" | Should -Match "ndk;$ndkPackage"
+        }
+
+        It "Default NDK is installed" -TestCases @(@{ ndkDefaultVersion = $ndkDefaultMajorVersion; installedPackages = $androidInstalledPackages }) {
+            "$installedPackages" | Should -Match "ndk;$ndkDefaultVersion"
```
Comment (MaksimZhukov) [inline images/win/scripts/Tests/Android.Tests.ps1:103]: Does it make sense to add a test to check that the ndk-bundle points to the default ndk version from toolset as we did for macOS and Ubuntu images?

Comment (nikita-bykov) [reply images/win/scripts/Tests/Android.Tests.ps1]: Added this test

**Review by MaksimZhukov — DISMISSED** [review]

File: images/win/scripts/Installers/Install-AndroidSDK.ps1 [hunk 5]
```diff
@@ -109,21 +111,23 @@ $ndkLatestPackageName = Get-AndroidPackagesByName -AndroidPackages $androidPacka
...
+    Get-AndroidPackagesByName -AndroidPackages $androidPackages -PrefixPackageName "ndk;$_" | Sort-Object -Unique | Select-Object -Last 1
+}
 
 Install-AndroidSDKPackages -AndroidSDKManagerPath $sdkManager `
                 -AndroidSDKRootPath $sdkRoot `
                 -AndroidPackages $androidNDKs
 
-$ndkLTSVersion = $ndkLTSPackageName.Split(';')[1]
+$ndkDefaultVersion = $ndkDefaultPackageName.Split(';')[1]
 $ndkLatestVersion = $ndkLatestPackageName.Split(';')[1]
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-AndroidSDK.ps1:123]: Looks a bit complicated. I think we can get rid of lines 104 and 109 as we have `$androidNDKs` variable now. These two would look like
```
$ndkDefaultVersion = ($androidNDKs | Where-Object { $_ -match "ndk;$ndkDefaultMajorVersion" }).Split(';')[1]
$ndkLatestVersion = ($androidNDKs | Where-Object { $_ -match "ndk;$ndkLatestMajorVersion" }).Split(';')[1]
```

Comment (nikita-bykov) [reply images/win/scripts/Installers/Install-AndroidSDK.ps1]: Updated it, thank you!

**Review by miketimofeev — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Pull request merged**